File: The Coupon Code.ts
export function checkCoupon(enteredCode: string, correctCode: string, currentDate: string, expirationDate: string): boolean {
  const d1 = new Date(currentDate);
  const d2 = new Date(expirationDate);
  return d1<=d2 && (enteredCode===correctCode)
}

File: The Coupon Code/The Coupon Code.py
import datetime
def check_coupon(entered_code, correct_code, current_date, expiration_date):
    d1 = datetime.datetime.strptime(current_date, "%B %d, %Y")
    d2 = datetime.datetime.strptime(expiration_date, "%B %d, %Y")
    return entered_code==correct_code and type(entered_code)==type(correct_code) and d1<=d2
